### PR TITLE
Add Samples for Vertex AI RAG

### PR DESCRIPTION
**Pull request by holtskinner**
Needed for cl/620709657

### CODE REVIEW

File: generative_ai/rag.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    # [END generativeaionvertexai_rag_upload_file]
+    return rag_file
+
+
+# [START generativeaionvertexai_rag_import_files]
+def import_files(
+    project_id: str,
+    location: str,
+    corpus_name: str,
+    path: Union[str, List[str]],
```
Comment (yinghsienwu) [inline generative_ai/rag.py:113]: paths: List[str],



(I changed this to address your comment)

File: generative_ai/rag.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    location: str,
+    rag_corpus: str,
+    text: str,
+    similarity_top_k: Optional[int] = 10,
+):
+    # Initialize Vertex AI
+    vertexai.init(project=project_id, location=location)
+
+    response = rag.retrieval_query(
+        rag_corpora=rag_corpus, text=text, similarity_top_k=similarity_top_k
```
Comment (yinghsienwu) [inline generative_ai/rag.py:192]: rag_corpor=[rag_corpus]


(I changed it to address your comment)

File: generative_ai/rag.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        corpus.name,
+        paths,
+        chunk_size=1024,
+        chunk_overlap=200,
+    )
+
+    rag_retrieval_tool = Tool.from_retrieval(
+        retrieval=rag.Retrieval(
+            source=rag.VertexRagStore(
+                rag_corpora=corpus.name,  # Only 1 corpus is allowed.
```
Comment (yinghsienwu) [inline generative_ai/rag.py:234]: [corpus.name]

File: generative_ai/rag.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+                similarity_top_k=3,
```
Comment (yinghsienwu) [inline generative_ai/rag.py:235]: similarity_top_k=3,
vector_distance_threshold=0.4,

Please add this new param "vector_distance_threshold"

File: generative_ai/rag.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        display_name=display_name,
+        description=description,
+    )
+    print(rag_file)
+    # [END generativeaionvertexai_rag_upload_file]
+    return rag_file
+
+
+# [START generativeaionvertexai_rag_import_files]
+def import_files(
```
Comment (yinghsienwu) [inline generative_ai/rag.py:109]: Could you add import_files_async sample as well?

File: generative_ai/requirements.txt [hunk 6]
```diff
@@ -2,6 +2,6 @@ pandas==1.3.5; python_version == '3.7'
 pandas==2.0.1; python_version > '3.7'
 pillow==9.5.0; python_version < '3.8'
 pillow==10.3.0; python_version >= '3.8'
-google-cloud-aiplatform[pipelines]==1.47.0
+google-cloud-aiplatform[pipelines]==1.48.0
```
Comment (yinghsienwu) [inline generative_ai/requirements.txt:5]: pending

**Review by yinghsienwu — COMMENTED** [review]
Thanks for the PR

**Comment by qizzzh** [comment]
Thanks Holt! small nit: `vertexai.init` has some overhead. Maybe we can have some comment saying it only needs to be called once per session or so.

**Comment by holtskinner** [comment]
@qizzzh

The code samples team is trying to simplify our samples and reduce boilerplate where needed. The other Gemini samples don't include anything about doing the init once per session, should they be changed as well?

**Comment by qizzzh** [comment]
I think some comments may be sufficient for now. It's not difficult for developers to figure it out.

**Comment by holtskinner** [comment]
Note: I've tested the samples locally with the dev SDK and an allowlisted project, and they are all passing.

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**